src/integrators/integrator_utils.jl

```diff
@@ -152,35 +152,96 @@ function solution_endpoint_match_cur_integrator!(integrator)
   end
 end
 
-### Default is PI-controller
 function stepsize_controller!(integrator,alg)
-  # PI-controller
-  EEst,beta1,q11,qold,beta2 = integrator.EEst, integrator.opts.beta1, integrator.q11,integrator.qold,integrator.opts.beta2
-  if iszero(EEst)
-    q = inv(integrator.opts.qmax)
-  else
-    @fastmath q11 = EEst^beta1
-    @fastmath q = q11/(qold^beta2)
-    integrator.q11 = q11
-    @fastmath q = max(inv(integrator.opts.qmax),min(inv(integrator.opts.qmin),q/integrator.opts.gamma))
+  if ispredictive(alg)
```

Comment:
this looks a bit harder to maintain, add to, and understand. Why not still have smaller functions that end up called, but inline them?